File: test/cpp/api/cuda.cpp
#include <gtest/gtest.h>

#include <torch/cuda.h>

TEST(CUDA, device_count) {
  // Shouldn't throw an error
  torch::cuda::device_count();
}

TEST(CUDA, is_available) {
  // Shouldn't throw an error
  torch::cuda::is_available();
}

File: torch/_inductor/codegen/cutlass/lib_extensions/cutlass_mock_imports/cuda/cuda.py
# mypy: disable-error-code="no-untyped-def"
# flake8: noqa
import torch


class CUdeviceptr:
    pass


class CUstream:
    def __init__(self, v):
        pass


class CUresult:
    CUDA_SUCCESS = True


class nvrtc:
    pass


def cuDeviceGetCount():
    return (CUresult.CUDA_SUCCESS, torch.cuda.device_count())
